docs/plugin-proposal-decorators.md

```diff
@@ -41,13 +63,20 @@ class C {
 }
 
 function enumerable(value) {
-  return function (target, key, descriptor) {
-     descriptor.enumerable = value;
-     return descriptor;
-  }
+  return function(elementDescriptor) {
+    return {
+      ...elementDescriptor,
+      descriptor: {
+        ...elementDescriptor.descriptor,
+        enumerable: value
+      }
```

Comment:
All this function can be simplified to
```js
return function(elementDescriptor) {
  elementDescriptor.descriptor.enumerable = value;
};
```

Reply:
I just noticed the examples of the latest proposal were wroten in immutable style. But it looks like not necessary.
So the others can also be simplified to
```js
function annotation(classDescriptor) {
  classDescriptor.elements.push({
    kind: 'field',
    key: 'annotated',
    placement: 'static',
    descriptor: {},
    initializer: () => true,
  });
}

function isTestable(value) {
  return function(classDescriptor) {
    classDescriptor.elements.push({
      kind: 'field',
      key: 'isTestable',
      placement: 'static',
      descriptor: {},
      initializer: () => value,
    });
  };
}
```
It works in Babel 7.1.0. However, the [proposal](https://tc39.github.io/proposal-decorators/#sec-decorator-functions) said a decorator function returns a decorator. So should I add a return ?
```js
return function(elementDescriptor) {
  elementDescriptor.descriptor.enumerable = value;
  return elementDescriptor
};
```

Reply:
If a decorator returns nothing, it returns the original descriptor (probably that introduction needs to be updated, in you want to open a PR).

I'd like to have examples in both styles, to show different ways of using decorators.